.github/workflows/python.yml

```diff
@@ -0,0 +1,198 @@
+name: Python
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - '*'
+    paths:
+      - python/**
+  pull_request:
+    paths:
+      - python/**
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.8"
+      - name: Build
+        uses: PyO3/maturin-action@v1
+        with:
+          command: develop
+          args: -m python/Cargo.toml
+      - name: Install dev requirements
+        run: pip install -r python/requirements-dev.txt
+      - name: Check
+        run: mypy python && ruff check python && ruff format python --check
```

Comment:
Nit: instead of using `python` everywhere, set the default working directory for this CI job to `python`? 
https://github.com/geoarrow/geoarrow-rs/blob/87992ccb5e2b78a3b9975f2e29afbc9fa3a63ccf/.github/workflows/python.yml#L17-L19